src/SerialFirmata.cpp

```diff
@@ -296,27 +310,43 @@ void SerialFirmata::checkSerial()
         continue;
       }
 #endif
-      if (serialPort->available() > 0) {
-        Firmata.write(START_SYSEX);
-        Firmata.write(SERIAL_MESSAGE);
-        Firmata.write(SERIAL_REPLY | portId);
-
-        if (bytesToRead == 0 || (serialPort->available() <= bytesToRead)) {
-          numBytesToRead = serialPort->available();
-        } else {
-          numBytesToRead = bytesToRead;
+      int availableBytes = serialPort->available();
+      if (availableBytes > 0) {
+        bool read = true;
+        
+#if defined(SERIAL_STORE_AND_FORWARD)    
+        // check if reading should be delayed to collect some bytes before forwarding (for baud rates up to 19200)
```

Comment:
Change text within parenthesis to "for rates up to 49999"  - that is if I'm understanding this correctly - or is it only good for 50000 / 2?